### ISSUE TITLE
Calling GSDRUNet disables gradient computations

### ISSUE THREAD

**Issue by johertrich** [deepinv/deepinv#166]
There is a bug for the GSDRUNet. Calling it disables gradient computations until `torch.set_grad_enabled(True)` is called the next time. Below a small example to reproduce the bug.

```
import deepinv
import torch

x = torch.tensor(12.,requires_grad=True)
y = x**2-2

print(y.requires_grad)
# returns True as expected

img = torch.rand((1,3,30,30))
model = deepinv.models.GSDRUNet(in_channels=3, out_channels=3)

out = model(img,0.1)

x = torch.tensor(12.,requires_grad=True)
y = x**2-2

print(y.requires_grad)
# returns False which is (at least for me) unexpected without
# looking into the code of GSDRUNet...
```

The reason for the bug is that the `potential_grad` method in GSPnP calls `torch.set_grad_enabled(False)` without enabling the gradients afterwards again.

**Comment by matthieutrs** [comment]
@johertrich thanks for noticing :) I think the above merge solved the issue, feel free to reopen if not the case
